File: app/src/main/java/darkpoint/rubel/APIFactory.java
package darkpoint.rubel;

import com.google.gson.Gson;
import com.google.gson.GsonBuilder;
import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public final class APIFactory {

    public static ServerAPI createApi() {

        Gson gson = new GsonBuilder().setDateFormat("yyyy-MM-dd'T'HH:mm:ss").create();

        Retrofit retrofit = new Retrofit.Builder()
                .baseUrl("http://www.nbrb.by/API/ExRates/")
                .addConverterFactory(GsonConverterFactory.create(gson))
                .build();

        return retrofit.create(ServerAPI.class);
    }
}

File: app/src/main/java/darkpoint/rubel/network/APIFactory.kt
package darkpoint.rubel.network

import com.jakewharton.retrofit2.adapter.kotlin.coroutines.CoroutineCallAdapterFactory
import com.squareup.moshi.Moshi
import com.squareup.moshi.kotlin.reflect.KotlinJsonAdapterFactory
import retrofit2.Retrofit
import retrofit2.converter.moshi.MoshiConverterFactory

object APIFactory {

    fun createApi(): ServerAPI {

        //val gson = GsonBuilder().setDateFormat("yyyy-MM-dd'T'HH:mm:ss").create()

        val retrofit = Retrofit.Builder()
                .baseUrl("http://www.nbrb.by/API/ExRates/")
                .addCallAdapterFactory(CoroutineCallAdapterFactory())
                .addConverterFactory(MoshiConverterFactory.create(moshi))
                .build()

        return retrofit.create(ServerAPI::class.java)
    }
}

private val moshi = Moshi.Builder()
        .add(KotlinJsonAdapterFactory())
        .build()